src/relax/op/op_common.h

```diff
@@ -33,32 +33,51 @@
 namespace tvm {
 namespace relax {
 
-bool EqualConstInt(const PrimExpr& lhs, int64_t value);
+#define RELAX_REGISTER_OP RELAY_REGISTER_OP
 
-bool EqualCheck(const PrimExpr& lhs, const PrimExpr& rhs);
+/*!
+ * \brief Check if the given PrimExpr is the given constant integer.
+ * \param e The input PrimExpr to be checked.
+ * \param value The input constant integer.
+ * \return A boolean indicating if the given PrimExpr is the given constant integer.
+ */
+inline bool IsConstInt(const PrimExpr& e, int64_t value) {
+  const int64_t* pvalue = tir::as_const_int(e);
+  return pvalue != nullptr && *pvalue == value;
+}
 
-#define RELAX_REGISTER_OP RELAY_REGISTER_OP
+/*!
+ * \brief Check if the input PrimExprs can be proved to be equal.
+ * \param lhs The left hand side PrimExpr.
+ * \param rhs The right hand side PrimExpr
+ * \return A boolean indicating if the input PrimExprs can be proved to be equal.
+ */
+inline bool CanProveEqual(const PrimExpr& lhs, const PrimExpr& rhs) {
+  tvm::arith::Analyzer ana;
```

Comment:
see comment on the other side, consider inline into the logic via Builder.GetAnalayzer()